packages/core/lib/console.js

```diff
@@ -143,16 +126,53 @@ class Console extends EventEmitter {
     abstractions.forEach(abstraction => {
       contextVars[abstraction.contract_name] = abstraction;
     });
+  }
+
+  runSpawn(inputStrings, options, callback) {
+    let args;
+    if (typeof BUNDLE_CONSOLE_CHILD_FILENAME !== "undefined") {
+      args = path.join(__dirname, BUNDLE_CONSOLE_CHILD_FILENAME);
+    } else {
+      args = path.join(__dirname, "../lib/console-child.js");
+    }
+
+    const spawnOptions = { stdio: ["inherit", "inherit", "inherit"] };
 
-    this.repl.setContextVars(contextVars);
+    try {
+      //serialize config networks entries
+      const configNetworks = {};
+      for (const key in options.networks) {
+        configNetworks[key] = JSON.stringify(options.networks[key]);
+      }
+
+      spawnSync(
+        "node",
+        [args, inputStrings, JSON.stringify({ networks: configNetworks })],
+        spawnOptions
+      );
+
+      try {
+        this.provision();
+      } catch (e) {
+        console.log(e);
+      }
+    } catch (err) {
+      callback(err);
+    }
```

Comment:
Is the intention to eat exceptions thrown by `this.provision()` and not invoke the callback on 160?

Reply:
Unless I'm misunderstanding, the callback on lime 160 is for any exceptions thrown by spawning the repl. There's a try/catch for `this.provision()` and a try/catch for the spawning. Does that make sense or am I not following?

Reply:
The intention is a bit hard to follow because the two catch expressions are next to each other.  As it exists, if the provision after spawn fails then the code explicitly eats the error and continues execution to the next REPL prompt. Is an exception thrown by `this.provision()` benign as the code suggests?

Reply:
`this.provision();` can throw an Error (line 110) if something is awry with an artifact file. I think we should be invoke the callback in the outer catch block.